README.md

```diff
@@ -1,50 +1,129 @@
-go build
-```
 
-### Create the network
+If you don't want slack alerts or if you don't want to deploy web services then you can omit the slack and SSL ENVs respectively. 
+
+> All the secrets are stored in 1password and tagged as go-spacecraft.
+> 
+
+## Using the CLI
 
-The final step is to create the network. Run the below command to create a devnet:
+You can download and compile spacecraft using the following commands:
 
+```jsx
+git clone git@github.com:spacemeshos/go-spacecraft.git
+cd go-spacecraft && go build
 ```
-./go-spacecraft createNetwork --config=./artifacts/devnet/config.json
+
+Now you can explore the CLI flags and sub-commands using:
+
+```jsx
+./go-spacecraft --help
 ```
 
-This command will take around ~10min to finish and at the end you should see the kibana URL. The generated config file is uploaded to GCP storage. 
+Now you can refer to [devnet deployment guide](docs/devnet.md) to understand how to use the CLI to deploy a new network and web services.
+
+## Network Deployment
+
+For creating a new network boostrap deploys one bootstrap node, multiple bootnodes and then multiple miners. The bootnodes connect to bootstrap for p2p discovery and miners connect to bootnodes for p2p discovery. 
+
+Spacecraft also allows you to create a network that actually extends another network i.e., deploys a network without bootstrap. In this case the bootnodes connect to the bootnodes of the other network and miners of this new network connect to the bootnodes of this new network. This can be useful when you want to delete all the managed miners and other deployments and create a fresh network that connects to an existing network. You can achieve this using the `--bootstrap=false` flag.
+
+Whenever a network is created it also deploys a N number of PoETs. The PoET are assigned to the nodes in round robin fashion. The first node is then added to the config file so home smeshers use the first PoET always. 
+
+The bootnodes are always tied to a particular k8s worker node in round robin fashion. There are mainly two reasons for doing this: 
+
+- Firstly if we don't do this then its possible k8s will host multiple bootnodes in one k8s worker node and if that VM crashes then all bootnodes are down and new smeshers cannot join the network.
+- Secondly the TCP port of the bootnodes needs to be bind'ed to the host node otherwise p2p discovery fails. p2p discover doesn't work over k8s NodePort service. Therefore if we don't bind bootnodes to host node then the k8s node in which the bootnodes are hosted can be rescheduled and therefore the IP to reach the bootnodes will change and smeshers will not be able to connect to the all or some bootnodes.
+
+Spacecraft calculates the total number of k8s nodes need to be created dynamically based on total pods and their resource size. 
+
+The GRPC and JSON API ports of all the managed miners are exposed as NodePort. You can get list of all the managed miners GRPC URLs using the `hosts` sub-command.
```

Comment:
If these are 2 different ports, how are they exposed as a single `NodePort`?

Reply:
Was this resolved? Still can't find info about it.

Reply:
Yes its resolved. I had added this line "The GRPC and JSON API ports of all the managed miners are exposed as separate NodePort. You can get list of all the managed miners GRPC URLs using the `hosts` sub-command."